icu4c/source/common/unicode/utfiterator.h

```diff
@@ -1010,19 +1063,73 @@ class UTFIterator {
 
     /**
      * @param other Another iterator
-     * @return true if this operator is at the same position as the other one
+     * @return true if this iterator is at the same position as the other one
      * @draft ICU 78
      */
     U_FORCE_INLINE bool operator==(const UTFIterator &other) const {
         return getLogicalPosition() == other.getLogicalPosition();
     }
     /**
      * @param other Another iterator
-     * @return true if this operator is not at the same position as the other one
+     * @return true if this iterator is not at the same position as the other one
      * @draft ICU 78
      */
     U_FORCE_INLINE bool operator!=(const UTFIterator &other) const { return !operator==(other); }
 
+    // Asymmetric equality & nonequality with a sentinel type.
+    // C++17: Need to define all four combinations of == / != vs. parameter order.
+    // Once we require C++20, we could remove all but the first == because
```

Comment:
You might want to `#if` this to make it abundantly clear what it is that can be removed in the future and already now verify that this works.

Reply:
Good idea. Done.